CMakeLists.txt

```diff
@@ -24,6 +24,11 @@ list(APPEND BUILT_WITH "${CMAKE_C_COMPILER_ID}")
 list(APPEND BUILT_WITH "${CMAKE_CXX_COMPILER_ID}")
 # }
 
+if (MINGW AND RDKAFKA_BUILD_STATIC)
+  # ".dll.a" takes precedence over ".a" by default
+  set(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
+endif()
```

Comment:
Is it really useful? I would say that changing default behavior should be avoided to follow rule of least surprise.

Reply:
It's required to make sure we link to the static libraries (`.a`) and not the dynamic libraries (`.dll.a`). the only other way to ensure this is to delete the `.dll.a` files from the `mingw64/lib` directory, which is slightly less than ideal.

Reply:
It's an environment consideration, CMakeLists shouldn't have this logic. It's the responsability of consumers to provide either static or shared libs. Build files of a library shouldn't force to link static or shared dependencies.

When librdkafka is static, nothing is linked to the static lib anyway, and there is no reason to force all static just because librdkafka is static.

Reply:
Actually for this librdkafka-static build we do want all dependencies to be available as static libraries so we can provide a self-contained library.

Reply:
Is there a reason for that?

Reply:
I mean in conan recipe of librdkafka, default is all static, but consumers can set any combination of shared/static for all the dependency graph, and I didn't notice any issue.

Reply:
Ah. Maybe some background would help. The primary purpose of this PR (MinGW static build) is to have Travis build a static library that can be bundled with confluent-kafka-go, so users of that library are not required to have librdkafka and/or its dependencies previously installed on their system. A C/C++ user of librdkafka would _very_ likely be using the NuGet librdkafka package or the vcpkg one. Both of those use dynamic linking, and the MinGW dynamically linked build also works well (thanks @edbordin).

Reply:
And if you pass `CMAKE_FIND_LIBRARY_SUFFIXES` value to cmake command line in CI files, instead of hardcoded value in CMakeLists?